### PR TITLE
create controllers

### PR DESCRIPTION
**Pull request by Tobby8629**
create users and posts controller
- create the 7Restful routes using resources
- use nested resources; nesting the post resources inside the user resources
- create the User controller and it view file
- create the post controller

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @Tobby8629, Kindly fix the linter errors 👍🏿

**Comment by Tobby8629** [comment]
i fixed them, thank you very much 🙏🏽

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+class PostsController < ApplicationController
+end
```
Comment (prolajumokeoni) [inline app/controllers/posts_controller.rb]: - [ ] In your posts_controller, please add  #index and #show actions, it's okay for the actions to be empty for now.

````ruby
 def index; end

  def show; end
````


- [ ] Kindly add corresponding view files for the index and show controller actions, for your posts controller 👍🏿

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
```
Comment (prolajumokeoni) [inline app/controllers/users_controller.rb:1]: - [ ] In your users_controller, please add  #show actions, it's okay for the actions to be empty for now.

````ruby
  def show; end
````

- [ ] Kindly add corresponding view files for the show controller actions, you currently have only the index view file for your users controller 👍🏿

File: config/routes.rb [hunk 3]
```diff
@@ -3,4 +3,8 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  resources :users do
+    resources :posts
+  end
 end
```
Comment (prolajumokeoni) [inline config/routes.rb]: - [ ] Kindly configure your routes properly, so you don't expose unnecessary endpoints

```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Tobby8629 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Controllers,  Views and Route files created 👏



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Tobby8629 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Controllers,  Views and Route files created 👏



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @Tobby8629 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Controllers,  Views and Route files created 👏



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class PostsController < ApplicationController
```
Comment (bobb-Rob) [inline app/controllers/posts_controller.rb:1]: - As mentioned by the previous reviewer, Kindly add corresponding view files for the index and show controller actions, for your posts and user controller 👍🏿. I noticed you added only index.html.erb for user. Please add the show HTML.erb for users and the both index.html.erb and show.html.erb for post.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️

Hi @Tobby8629, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Reem-lab — APPROVED** [review]
Hi @Tobby8629, 🤩🔥 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
- [ ] Great job on the functionality for set up 🚀
- [ ] Great job working in the routes✨✨
- [ ] Professional README 🔥
- [ ] Followed Gitflow 🔥

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**